datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/Utils.scala

```diff
@@ -519,4 +521,73 @@ object Utils extends PredicateHelper {
     }
   }
 
+  /**
```

Comment:
Here better to add a comment, which explains the structure of the expression tree, first write the expression and better to draw a tree, of ScalaUDFExpr, children as TimeSeriesFun and Literal , so it will hep to understand the below logic in a better way.